### PR TITLE
Add tentative Generator ABC implementing ask and tell, accepting and returning lists of dictionaries

### PR DESCRIPTION
**Pull request by jlnav**

### CODE REVIEW

File: generator.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            [{"x": 1, "y": 1}, {"x": 2, "y": 2}, {"x": 3, "y": 3}]
+        """
+
+    def tell(self, results: List[dict], *args, **kwargs) -> None:
+        """
+        Send the results of evaluations to the generator.
+        
+        .. code-block:: python
+        
+            >>> results = [{"f": 1}, {"f": 4}, {"f": 9}]
```
Comment (RemiLehe) [inline generator.py:55]: This should also contain the corresponding `"x"` and `"y"` values, no?

File: generator.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        Initialize the Generator object on the user-side. Constants, class-attributes,
+        and preparation goes here.
+
+        .. code-block:: python
+
+            >>> my_generator = MyGenerator(my_parameter, my_keyword=10)
+        """
+
+    @abstractmethod
+    def ask(self, num_points: Optional[int], *args, **kwargs) -> List[dict]:
```
Comment (RemiLehe) [inline generator.py:38]: Do we need `*args, **kwargs` here?

Comment (RemiLehe) [inline generator.py:38]:
```suggestion
    def ask(self, num_points: Optional[int]) -> List[dict]:
```

File: generator.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            [{"x": 1, "y": 1}, {"x": 2, "y": 2}, {"x": 3, "y": 3}]
+        """
+
+    def tell(self, results: List[dict]) -> None:
+        """
+        Send the results of evaluations to the generator.
+        
+        .. code-block:: python
+        
+            >>> results = [{"f": 1}, {"f": 4}, {"f": 9}]
```
Comment (RemiLehe) [inline generator.py:55]:
```suggestion
            >>> results = [{"x": 0.5, "y": 1.5, "f": 1}, {"x": 2, "y": 3, "f": 4}]
```

**Comment by jlnav** [comment]
We want to pull this in? It may be a tad easier for us to experiment with it on main than my fork. We can always keep iterating later on

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**